### PR TITLE
17 add feature to update a pronamic pay subscription from a restrict content membership

### PR DESCRIPTION
**Pull request by remcotolsma**
This PR adds a "Update" button on the Restrict Content "Membership Details" page in the WordPress admin dashboard:

Scherm­afbeelding 2024-05-07 om 13 22 11

This allows administrators to update the Restrict Content "Membership Details" to the Pronamic Pay subscription.

### CODE REVIEW

**Comment by remcotolsma** [comment]
Another idea, isn't it better and easier to hook this functionality into the Restrict Content Pro button "Update Membership" button? @rvdsteege

**Comment by rvdsteege** [comment]
Wouldn't that be conflicting with the RCP warning message when editing the expiration date?

> Changing the expiration date will not affect when renewal payments are processed.

**Comment by remcotolsma** [comment]
Oh yeah, that's why I set it up like that. 😅 But this message does not necessarily have to be shown:

- https://plugins.trac.wordpress.org/browser/restrict-content/tags/3.2.10/core/includes/admin/memberships/edit-membership.php#L331
- https://plugins.trac.wordpress.org/browser/restrict-content/tags/3.2.10/core/includes/memberships/class-rcp-membership.php#L3454

Adjust the PR? @rvdsteege

File: src/Extension.php [hunk 1]
```diff
@@ -443,11 +445,15 @@ public function subscription_status_update( Subscription $pronamic_subscription
...
-			return;
+		if ( null !== $status ) {
+			$rcp_membership->set_status( $status );
 		}
 
-		$rcp_membership->set_status( $status );
+		$id = $pronamic_subscription->get_id();
+
+		if ( null !== $id ) {
+			$rcp_membership->set_gateway_subscription_id( (string) $id );
```
Comment (rvdsteege) [inline src/Extension.php:455]: Discussed with @remcotolsma at office, setting the gateway subscription ID is required to show the correct messages when editing a membership.

**Review by rvdsteege — APPROVED** [review]
Approved, but maybe also nice to hook the `rcp_gateway_subscription_id_url` filter to make the gateway subscription ID link work (instead of linking to the same page)?

https://plugins.trac.wordpress.org/browser/restrict-content/tags/3.2.10/core/includes/admin/memberships/edit-membership.php#L281

**Pull request merged**